packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts

```diff
@@ -30,21 +32,46 @@ const treeItemBehavior: Accessibility<TreeItemBehaviorProps> = props => ({
       performClick: {
         keyCombinations: [{ keyCode: keyboardKey.Enter }, { keyCode: keyboardKey.Spacebar }],
       },
-      collapseOrReceiveFocus: {
-        keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
-      },
-      expandOrPassFocus: {
-        keyCombinations: [{ keyCode: keyboardKey.ArrowRight }],
-      },
+      ...(isSubtreeOpen(props) && {
+        receiveFocus: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
+          shouldHandle: doesEventComesFromChildItem,
+        },
+        collapse: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
+          shouldHandle: event => !doesEventComesFromChildItem(event),
+        },
+      }),
+      ...(!isSubtreeOpen(props) && {
+        expand: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowRight }],
+        },
+      }),
+      ...(isSubtreeOpen(props) && {
+        passFocus: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowRight }],
+        },
+      }),
     },
   },
 })
 
+export default treeItemBehavior
+
 export type TreeItemBehaviorProps = {
   /** If item is a subtree, it contains items. */
   items?: object[]
   /** If item is a subtree, it indicates if it's open. */
   open?: boolean
 }
 
-export default treeItemBehavior
+/** Checks if the event comes from a child item */
+const doesEventComesFromChildItem = (event): boolean => {
+  return event.currentTarget !== event.target
+}
+
+/** Checks if current tree item has a subtree and it is opened */
+const isSubtreeOpen = (props: TreeItemBehaviorProps): boolean => {
+  const { items, open } = props
+  return !!(items && items.length && open)
```

Comment:
Consider if it is enough to check the `open` prop.  If so, consider if we can collapse this function to inline logic.